finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/TlsServerConfig.scala

```diff
@@ -0,0 +1,47 @@
+package com.twitter.finagle.buoyant
+
+import com.twitter.finagle.Stack
+import com.twitter.finagle.ssl.{ClientAuth => FClientAuth, _}
+import com.twitter.finagle.ssl.server.{SslServerConfiguration, SslServerEngineFactory}
+import com.twitter.finagle.transport.Transport
+import java.io.File
+
+case class TlsServerConfig(
+  certPath: String,
+  keyPath: String,
+  caCertPath: Option[String] = None,
+  ciphers: Option[Seq[String]] = None,
+  requireClientAuth: Option[Boolean] = None
+) {
+  def params(
+    alpnProtocols: Option[Seq[String]],
+    sslServerEngine: SslServerEngineFactory
```

Comment:
You're correct we could avoid passing it through. The call sites are different enough where I thought this approach minimized code complexity, since we already are passing `aplnProtocols` through anyway.